File: example/java/jimmer-sql2/src/main/java/org/babyfish/jimmer/sql2/example/model/Chapter.java
package org.babyfish.jimmer.sql2.example.model;

import org.babyfish.jimmer.sql.*;
import org.babyfish.jimmer.sql2.example.model.common.BaseEntity;
import org.jetbrains.annotations.Nullable;

@Entity
public interface Chapter extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    @Nullable
    @ManyToOne(inputNotNull = true)
    @Key
    @OnDissociate(DissociateAction.DELETE)
    Book book();

    @Key
    @Column(name = "chapter_no")
    int index();

    String title();
}

File: example/kotlin/jimmer-sql-static-kt/src/main/kotlin/org/babyfish/jimmer/sql/example/model/Chapter.kt
package org.babyfish.jimmer.sql.example.model

import org.babyfish.jimmer.sql.*
import org.babyfish.jimmer.sql.example.model.common.BaseEntity

@Entity
interface Chapter : BaseEntity {

    @Id
    val id: Long

    @Key
    @ManyToOne(inputNotNull = true)
    @OnDissociate(DissociateAction.DELETE)
    val book: Book?

    @Key
    @Column(name = "chapter_no")
    val index: Int

    val title: String
}